tools/generate-mocks/generate-mocks.ts

```diff
@@ -92,36 +140,72 @@ export const MOCK_COMPONENTS = [
+    exportedProviders: ComponentMetaData[]
   ) {
     const folderContent = readdirSync(folderpath);
     for (const fileOrFolder of folderContent) {
       const fullPath = path.join(folderpath, fileOrFolder);
       const ent = statSync(fullPath);
       if (ent.isDirectory()) {
-        await this.traverseFolder(fullPath, outputPath, subPath, classMap, exportedComponents);
+        await this.traverseFolder(fullPath, outputPath, classMap, exportedTypes, exportedProviders);
       } else {
         if (fileOrFolder.endsWith('.component.ts')) {
           // console.log('Rendering mock for: ', fullPath);
-          const newFilename = path.join(outputPath, subPath, 'mock.' + fileOrFolder);
-          const classNames = this.renderMock(fullPath, newFilename, exportedComponents);
+          const newFilename = path.join(outputPath, 'components', 'mock.' + fileOrFolder);
+          const classNames = this.renderMock(fullPath, newFilename, exportedTypes);
           if (classNames) {
             classMap.set(newFilename, classNames);
           }
         }
+        if (fileOrFolder.endsWith('.controller.ts') || fileOrFolder.endsWith('.service.ts')) {
+          exportedProviders.push(...this.getExportedProvidersMetadata(fullPath, exportedTypes));
+        }
       }
     }
   }
 
-  private renderMock(fileName: string, newFilename: string, exportedComponents: string[]) {
+  private getExportedProvidersMetadata(fileName: string, exportedTypes: string[]) {
+    return this.generateMetaData(fileName).filter(
+      (metaData) =>
+        metaData.decorator === 'Injectable' && exportedTypes.includes(metaData.className)
+    );
+  }
+
+  private renderJasmineMockProviderFactory(componentMetaData: ComponentMetaData) {
+    const methodNames = componentMetaData.methods.map((m) => `'${m.name}'`).join(',' + newLine);
+    const funcName = this.getMockProviderFactoryName(componentMetaData.className);
+    return `export function ${funcName}() {
+  return jasmine.createSpyObj('${componentMetaData.className}', [
+    ${methodNames}
+  ]); 
+}`;
+  }
+
+  private renderJestMockProviderFactory(componentMetaData: ComponentMetaData) {
+    const mockMethods = componentMetaData.methods
+      .map((m) => `${m.name}: jest.fn()`)
+      .join(',' + newLine);
+    const funcName = this.getMockProviderFactoryName(componentMetaData.className);
+    return `export function ${funcName}() {
+  return {
+    ${mockMethods}
+  };
+}`;
+  }
+
+  private getMockProviderFactoryName(className: string) {
+    return className[0].toLowerCase() + className.slice(1) + 'Factory';
+  }
+
+  private renderMock(fileName: string, newFilename: string, exportedTypes: string[]) {
     const components = this.generateMetaData(fileName);
-    if (!components.find((metaData) => metaData.decorator)) {
-      // Nothing to generate:
-      return;
-    }
-    if (!components.some((metaData) => exportedComponents.indexOf(metaData.className) > -1)) {
+    if (
+      !components.some(
```

Comment:
consider extracting this expression to a well-named method

Reply:
Fixed 👍🏻